pkg/domain/infra/abi/play.go

```diff
@@ -267,12 +267,9 @@ func (ic *ContainerEngine) playKubePod(ctx context.Context, podName string, podY
 	}
 
 	if podOpt.Infra {
-		imagePull := config.DefaultInfraImage
-		if podOpt.InfraImage != config.DefaultInfraImage && podOpt.InfraImage != "" {
-			imagePull = podOpt.InfraImage
-		}
+		containerConfig := util.DefaultContainerConfig()
```

Comment:
Doesn't this return an empty config? Why are we ignoring the podOpt.InfraImage field. If we are not using it anymore should we remove it? I feel like for consistency we should make that option work like the others.

Reply:
No, this will return the `infra_image` specified in containers.conf file, if no `infra_image` in containers.conf, it will return the default one.

Logics: https://github.com/containers/common/blob/854253d/pkg/config/config.go#L1008-L1015

https://github.com/containers/common/blob/854253d/pkg/config/default.go#L328

Reply:
the issue is (i think) eliminating the usage of `podOpt.InfraImage` messes up the ability to specify an infra image not in containers.conf

Reply:
I think the priority is:  `command line infra image flag > containers.conf > default one`.

I can submit another PR to allow setting infra image in command line, but I think users will lessly use `podman play kube --infra-image=xxx  pod.yaml`.